pkg/proto/nbcontract/v1/utils.go

```diff
@@ -0,0 +1,289 @@
+func GetNetworkPluginType(networkPlugin string) NetworkPlugin {
+	if strings.EqualFold(networkPlugin, "azure") {
+		return NetworkPlugin_NP_AZURE
+	} else if strings.EqualFold(networkPlugin, "kubenet") {
+		return NetworkPlugin_NP_KUBENET
+	}
+
+	return NetworkPlugin_NP_NONE
+}
+
+// GetNetworkPolicyType returns the NetworkPolicyType enum based on the input string.
+func GetNetworkPolicyType(networkPolicy string) NetworkPolicy {
+	if strings.EqualFold(networkPolicy, "azure") {
+		return NetworkPolicy_NPO_AZURE
+	} else if strings.EqualFold(networkPolicy, "calico") {
+		return NetworkPolicy_NPO_CALICO
+	}
+
+	return NetworkPolicy_NPO_NONE
+}
+
+// GetDefaultOutboundCommand returns a default outbound traffic command.
+func GetDefaultOutboundCommand() string {
+	return "curl -v --insecure --proxy-insecure https://mcr.microsoft.com/v2/"
+}
+
+func GetKubeletNodeLabels(agentPool *datamodel.AgentPoolProfile) map[string]string {
+	kubeletLabels := map[string]string{
+		"agentpool":                      agentPool.Name,
+		"kubernetes.azure.com/agentpool": agentPool.Name,
+	}
+	for key, val := range agentPool.CustomNodeLabels {
+		kubeletLabels[key] = val
+	}
+	return kubeletLabels
+}
+
+// NOTE: The following functions are slightly modified versions of those that are already in agent/utils.go.
+// Other functions from agent/utils.go will also need to be added/merged here.
+
+// GetOutBoundCmd returns a proper outbound traffic command based on some cloud and Linux distro configs.
+func GetOutBoundCmd(nbc *datamodel.NodeBootstrappingConfiguration) string {
+	cs := nbc.ContainerService
+	if cs.Properties.FeatureFlags.IsFeatureEnabled("BlockOutboundInternet") {
+		return ""
+	}
+
+	if strings.EqualFold(nbc.OutboundType, datamodel.OutboundTypeBlock) || strings.EqualFold(nbc.OutboundType, datamodel.OutboundTypeNone) {
+		return ""
+	}
+
+	var registry string
+	switch {
+	case nbc.CloudSpecConfig.CloudName == datamodel.AzureChinaCloud:
+		registry = `gcr.azk8s.cn`
+	case cs.IsAKSCustomCloud():
+		registry = cs.Properties.CustomCloudEnv.McrURL
+	default:
+		registry = `mcr.microsoft.com`
+	}
+
+	if registry == "" {
+		return ""
+	}
+
+	// curl on Ubuntu 16.04 (shipped prior to AKS 1.18) doesn't support proxy TLS.
+	// so we need to use nc for the connectivity check.
+	clusterVersion, _ := semver.Make(cs.Properties.OrchestratorProfile.OrchestratorVersion)
+	minVersion, _ := semver.Make("1.18.0")
+
+	var connectivityCheckCommand string
+	if clusterVersion.GTE(minVersion) {
+		connectivityCheckCommand = `curl -v --insecure --proxy-insecure https://` + registry + `/v2/`
+	} else {
+		connectivityCheckCommand = `nc -vz ` + registry + ` 443`
+	}
+
+	return connectivityCheckCommand
+}
+
+// GetOrderedKubeletConfigFlagString returns an ordered string of key/val pairs.
+// copied from AKS-Engine and filter out flags that already translated to config file.
+func GetKubeletConfigFlag(k map[string]string, cs *datamodel.ContainerService, profile *datamodel.AgentPoolProfile,
```

Comment:
Just to confirm, this function is not really used in the current PR, right? Is it supposed to be used by AKS-RP later?

Reply:
see above